### ISSUE TITLE
Very high classification accuracy, very low regression accuracy

### ISSUE THREAD

**Issue by LaurensHagendoorn** [fizyr/keras-retinanet#83]
After having trained the network (which seemed to go fine) I am not getting any predictions on the test set (all predictions are 0.01). When determining the accuracy on the train set even, I am getting a very high classification accuracy and a very poor regression accuracy.
My guess is that since it doesn't find any correct bounding boxes, the classification is very good. This is of course not the intended behavior.
Any ideas as to what might be going wrong?

loss: 19.55561057557451
regression_loss: 18.411668621545942
classification_loss: 1.1439420243794014
regression_acc: 0.22899062909845008
classification_acc: 0.9996446640066978

**Comment by hgaiser** [comment]
Hmm how can the predictions all be `0.01` while you get high classification accuracy? If the box regression failed, it should still give those badly regressed boxes with high classification accuracy since they are decoupled in the training process.

**Comment by LaurensHagendoorn** [comment]
is it? I figured if it didn't find any boxes it has nothing to classify? And I was indeed also quite confused by this result.
I am using a batchsize of 8 during training, and of 1 during testing, but that should not be the problem right?

**Comment by hgaiser** [comment]
If you use the [changes](https://github.com/fizyr/keras-retinanet/commit/1049d0644db07c6bddc82adf1b3f38d4313521f2) from this morning it should (theoretically) not be a problem. If you want to be sure, try training with batch_size = 1.

**Comment by LaurensHagendoorn** [comment]
Ah, I do not use the changes from this morning, let's hope that's it!

**Comment by hgaiser** [comment]
I fear they would only improve your results, but not explain why you get no results at all.

**Comment by LaurensHagendoorn** [comment]
I will also try training with a batch size of 1, we'll see. I checked if it loads the annotations, but those look good.

**Comment by RaananHadar** [comment]
@LaurensHagendoorn 'classic' accuracy doesn't mean much when it comes to object detection problem. Since your positive class is very rare and the majority of error comes from either missclassified positives or false alarms (which should be rare compared to the clutter) this means that accuracy of >99% is highly expected and non informative. What you care about is not an accuracy metric but a conditional accuracy metric over the positive class. 

Bottom line: you need to look at things like precision and recall or Pd and PFA to see how well your algorithm is doing.
